packages/@aws-cdk/aws-ecs/lib/base/pipeline-action.ts

```diff
@@ -0,0 +1,75 @@
+import codepipeline = require('@aws-cdk/aws-codepipeline-api');
+import { Repository } from '@aws-cdk/aws-ecr';
+import { PolicyStatement } from '@aws-cdk/aws-iam';
+import cdk = require('@aws-cdk/cdk');
+import { BaseService } from './base-service';
+
+/**
+ * Common properties for the {@link PipelineSourceAction CodePipeline deploy Action},
+ * whether creating it directly,
+ * or through the {@link IRepository#toCodePipelineSourceAction} method.
+ */
+export interface PipelineDeployActionProps extends codepipeline.CommonActionProps {
+  /**
+   * the input artifact contains the image definitions file which describes the ECR docker image(s) to deploy
+   * @see https://docs.aws.amazon.com/codepipeline/latest/userguide/pipelines-create.html#pipelines-create-image-definitions
+   */
+  inputArtifact: codepipeline.Artifact;
+
+  /**
+   * the filename of the image definitions file in the input artifact
```

Comment:
Can you use full sentences for the JSDocs (begins with an uppercase letter, ends with a period)? Thanks!